bin/header_scraper.py

```diff
@@ -110,55 +110,82 @@ def main():
                                 header = libraries[name]
                         continue
 
-                    # data[0] will be the function, macro, etc.
-                    # data[1] will be the description and type (e.g. "function", "macro constant", etc.)
+                    # data[0] is the function, macro, or typedef (herein an "element")
+                    # data[1] is the description and type (e.g. "function", "macro constant", etc.)
                     type = data[1].get_text()
+
                     # get span element containing all entries in this data block
-                    span = data[0].find("span")
-                    if not span:
+                    # skip over empty data blocks
+                    parent_span = data[0].find("span")
+                    if not parent_span:
                         continue
-                    # iterate over individual elements' spans
+
+                    # each row could contain multiple elements
+                    # in which case, they are all the same type and version, but should be separate rows in the CSV
+                    # create a string of all elements, later to be converted into a list
                     text = ""
-                    for element in span.find_all("span"):
-                        text += element.get_text().strip()
+                    for element in parent_span.find_all("span"):
+                        text += element.get_text().strip() + "\n"
 
-                    # add version of C this was added in, if appropriate
+                    # add C standard to which this element conforms, if given
                     version = data[0].find("span", class_="t-mark-rev")
                     if version:
-                        text += " " + version.get_text()
+                        # remove parentheses
+                        version = version.get_text().replace("(", "").replace(")", "")
+
+                    # make a tuple for each element, all should be same type
+                    for element in text.splitlines():
+                        el_tuple = tuple()
+                        el_tuple += (element,)
+
+                        # add in version if it was given
+                        if version:
+                            # "until" and "since" lines should go in 3rd column of CSV
+                            if "since" in version or "until" in version:
```

Comment:
Not quite, second column should represent "since" (i.e., when added), and third column should represent "until" (i.e., when removed). No words/chars other than C## should appear in each of those columns.